package.json

```diff
@@ -26,6 +26,7 @@
   },
   "scripts": {
     "build": "gulp build",
+    "full-test": "nyc mocha -r ts-node/register tests/**/*.spec.ts tests/**/**/*.spec.ts tests/**/**/**/*.spec.ts tests/**/*.e2e.ts tests/**/**/*.e2e.ts tests/**/**/**/*.e2e.ts",
```

Comment:
Unless I'm missing something, you only need to glob once for each pattern as `/**/` with `-r` says 'recursively descend the tree matching all suffix patterns`:
```json
"full-test": "nyc mocha -r ts-node/register tests/**/*.spec.ts tests/**/*.e2e.ts"
```

Reply:
Prior wisdom has me think this is not the case, will make this citable.

Reply:
![tests2](https://user-images.githubusercontent.com/20857369/39468950-8d15c1b4-4d2d-11e8-950e-f24de6118989.png)
![tests](https://user-images.githubusercontent.com/20857369/39468951-8d31781e-4d2d-11e8-9510-634855f86ad6.png)

Reply:
This confused me for a couple of minutes, because I couldn't see very well what was going on. A short text explanation would have done so much more for me.

So, the standard pattern for globbing in the shell is to quote them. I should have double-checked what I did instead of C&Ping. This should work fine for example:

```
nyc mocha -r ts-node/register 'tests/**/*.spec.ts'
```